js/ui/file_manager/ui.file_manager.item_list.js

```diff
@@ -0,0 +1,57 @@
+import { extend } from "../../core/utils/extend";
+
+import Widget from "../widget/ui.widget";
+
+const FILE_MANAGER_FILES_VIEW_CLASS = "dx-filemanager-files-view";
+
+class FileManagerItemListBase extends Widget {
+
+    _initMarkup() {
+        this.$element().addClass(FILE_MANAGER_FILES_VIEW_CLASS);
+    }
+
+    _getDefaultOptions() {
+        return extend(super._getDefaultOptions(), {
+            selectionMode: "single",
+            onGetItems: null,
+            onError: null,
+            onSelectedItemOpened: null,
+            getItemThumbnail: null
+        });
+    }
+
+    _getItems() {
+        const getItemsMethod = this.option("onGetItems");
+        return getItemsMethod();
```

Comment:
If the getItemsMethod const has null or undefined then an exception will be thrown.

Reply:
And where a test on this?

Reply:
This widget is internal, and it cannot work without this `onGetItems` option. This option is always specified in the calling code. In case when it is not set, it’s better to know about it with an exception.
We will add test for this options later.